Spatial.gd

```diff
@@ -641,15 +641,21 @@ func gen_new_icon(position: Vector3, texture_path: String, waypoint_icon, catego
 ################################################################################
 # Section of functions for saving changes to markers
 ################################################################################
-func on_change_made():
-	self.unsaved_data_icon.visible = true
+func update_burrito_icon():
+	if self.unsaved_changes:
+		$Control/GlobalMenuButton/TextureRect.modulate = Color(.78, 0.3, 0.3, 1.0)
+		$Control/GlobalMenuButton/main_menu_toggle.hint_tooltip = "Unsaved Data"
+	else:
+		$Control/GlobalMenuButton/TextureRect.modulate = Color(1.0, 1.0, 1.0, 1.0)
+		$Control/GlobalMenuButton/main_menu_toggle.hint_tooltip = ""
 
 func save_current_map_data():
 	var packed_bytes = self.waypoint_data.to_bytes()
-	if packed_bytes.size() > 0:
-		var file = File.new()
-		file.open(self.marker_file_path, file.WRITE)
-		file.store_buffer(packed_bytes)
+	var file = File.new()
+	file.open(self.marker_file_path, file.WRITE)
+	file.store_buffer(packed_bytes)
+	self.unsaved_changes = false
+	update_burrito_icon()
```

Comment:
How easy/likely is it for a programming mistake to lead to a desynced status and saved notification? EG: updating the variable without calling update_burrito_icon()